client/my-sites/hosting/github/github-connect-card/index.tsx

```diff
@@ -1,23 +1,132 @@
+	const [ selectedBranch, setSelectedBranch ] = useState< string >();
+	const { data: repos, isLoading: isLoadingRepos } = useGithubRepos( siteId );
+	const { data: branches, isLoading: isLoadingBranches } = useGithubBranches(
+		siteId,
+		selectedRepo,
+		{
+			onSuccess( branches ) {
+				setSelectedBranch( branches[ 0 ] );
+			},
+		}
+	);
+
+	const { connectBranch, isLoading: isConnecting } = useGithubConnectMutation( siteId, {
+		onSuccess: () => {
+			dispatch( successNotice( __( 'Branch connected successfully' ), noticeOptions ) );
+		},
+		onError: ( error ) => {
+			dispatch(
+				errorNotice(
+					// translators: "reason" is why connecting the branch failed.
+					sprintf( __( 'Failed to connect: %(reason)s' ), { reason: error.message } ),
+					{
+						...noticeOptions,
+					}
+				)
+			);
+		},
+	} );
+
+	const showSpinner = ! repos && ! branches && ( isLoadingRepos || isLoadingBranches );
+	const busy = isLoadingBranches || isConnecting;
+	const disabled = ! selectedBranch || ! selectedRepo;
+	const handleRepoSelect = ( repoName: string ) => {
+		setSelectedRepo( repoName );
+		setSelectedBranch( '' );
+	};
+
+	const handleBranchSelect = ( branchName: string ) => {
+		setSelectedBranch( branchName );
+	};
+
 	return (
-		<Card className="github-hosting-card">
-			<img className="github-hosting-icon" src={ iconGitHub } alt="" />
-			<CardHeading>{ __( 'Connect Branch' ) }</CardHeading>
-			<DisconnectGitHubExpander connection={ connection } />
+		<Card className="connect-branch-card">
+			<SocialLogo className="material-icon" icon="github" size={ 32 } />
+			<CardHeading>{ __( 'Deploy from GitHub' ) }</CardHeading>
+			<div>
+				<p>
+					{ __( 'Changes pushed to the selected branch will be automatically deployed. ' ) }
+					<ExternalLink href="#" target="_blank" rel="noopener noreferrer">
+						{ __( 'Learn more' ) }
+					</ExternalLink>
+				</p>
+				{ showSpinner ? (
+					<Spinner />
+				) : (
+					<FormFieldset className="connect-branch__fields">
+						<FormFieldset className="connect-branch__field">
+							<FormLabel htmlFor="repository">{ __( 'Repository' ) }</FormLabel>
+							<Search
+								id="repository"
+								placeholder={ __( 'Start typing a repo..' ) }
+								options={ repos ?? [] }
+								onSelect={ handleRepoSelect }
+							/>
+						</FormFieldset>
+						<FormFieldset className="connect-branch__field">
+							<FormLabel htmlFor="branch">{ __( 'Branch to deploy' ) }</FormLabel>
+							<FormSelect
+								className="connect-branch__field"
+								onChange={ ( event ) => handleBranchSelect( event.currentTarget.value ) }
+								value={ selectedBranch }
+							>
+								{ branches?.map( ( branch ) => (
+									<option value={ branch } key={ branch }>
+										{ branch }
+									</option>
+								) ) }
+							</FormSelect>
+						</FormFieldset>
+					</FormFieldset>
+				) }
+			</div>
+			<div style={ { marginTop: '10px', display: 'inline-flex', gap: '16px' } }>
```

Comment:
Would be nice to have the disconnect button on the other side of the card so the user won't misclick it.

Reply:
I tried but the UI looks unbalance.
![image](https://user-images.githubusercontent.com/47489215/219136193-3a2e0482-37b9-4857-af1a-87dd76d2b7e9.png)

Reply:
It does 😢

Can we try moving the CTA to the top-right corner of the screen and move the disconnect button to the bottom left?

Another alternative would be to flip the buttons (move disconnect to the left). Or, keep it on the right, but remove the border and leave it as red, just text -- the `transparent` variant

Or, if we're feeling lucky, we can just get rid of this button altogether 🤷‍♂️  I can't imagine a scenario where a user would need to totally disconnect GitHub. They might disconnect the hook but un-authorizing GitHub? Can't think of it.

Reply:
cc @javierarce @mpkelly would like some input on the above ☝️

Reply:
![image](https://user-images.githubusercontent.com/6851384/219252107-baab74f6-a725-472d-a9fb-fc55b74d5840.png)

I like this new UI 👍 . Visually, I think the buttons are ok as they are. But pinning them in the corners also works for me. 

We can improve it by showing that a query is in progress and by explicitly showing that no results were returned If you type "not-a-repo-name", it just sits there and is no visual feedback. 

>Another alternative would be to flip the buttons

On this page, the other cards have the CTA on the bottom left side. 

>Or, if we're feeling like it, we can just get rid of this button altogether 🤷‍♂️ 

I like having a complete reset option. The user can modify stuff at GitHub that could break this feature, so we should have a way to reset it. I agree with @javierarce on Slack that it should be more difficult to disconnect once you've connected a repo.